core/src/main/java/io/grpc/internal/GzipInflatingBuffer.java

```diff
@@ -199,6 +201,9 @@ boolean hasPartialData() {
    */
   void addGzippedBytes(ReadableBuffer buffer) {
     checkState(!closed, "GzipInflatingBuffer is closed");
+    if (buffer.readableBytes() > 0) {
```

Comment:
How about:
```
if (buffer.readableBytes() == 0) {
  return;
}
isStalled = false
```

Or just ignoring the case (everything should still function correctly, just be slightly less efficient. But we don't really expect that to happen, so no point optimizing.)

Reply:
Done